algebra/extractor/local_suprema.py

```diff
@@ -0,0 +1,87 @@
+from .. import Variant
```

Comment:
There should be a module docstring explaining what a local supremal representation is.